### ISSUE TITLE
No services detected on Windows machine

### ISSUE THREAD

**Issue by aleksandra-majchrzak** [jmdns/jmdns#133]
Hi! 
I created a simple project basing on the example in your README:

    public class MainClass {

      public static void main(String[] args) {
          try {
              // Create a JmDNS instance
              JmDNS jmdns = JmDNS.create(getIpAddress());

              // Add a service listener
              jmdns.addServiceListener("_http._tcp.local.", new SampleListener());

              // Wait a bit
              Thread.sleep(300);

              ServiceInfo serviceInfo = ServiceInfo.create("_http._tcp.local.", "example", 4567, "path=index.html");
              jmdns.registerService(serviceInfo);

          } catch (UnknownHostException e) {
              System.out.println(e.getMessage());
          } catch (IOException e) {
              System.out.println(e.getMessage());
          } catch (InterruptedException e) {
              e.printStackTrace();
          }
      }

      private static InetAddress getIpAddress() {
          try {
              for (Enumeration en = NetworkInterface.getNetworkInterfaces(); en.hasMoreElements(); ) {
                  NetworkInterface intf = (NetworkInterface) en.nextElement();
                  for (Enumeration enumIpAddr = intf.getInetAddresses(); enumIpAddr.hasMoreElements(); ) {
                      InetAddress inetAddress = (InetAddress) enumIpAddr.nextElement();
                     if (!inetAddress.isLoopbackAddress() && inetAddress instanceof Inet4Address
                              && intf.getName().contains("wlan")) {
                          String ipAddress = inetAddress.getHostAddress();
                         System.out.println("IP address: " + ipAddress);
                          return inetAddress;
                      }
                  }
              }
          } catch (SocketException ex) {
              System.out.println("Socket exception in GetIP Address of Utilities: " + ex.toString());
          }
          return null;
      }

      static class SampleListener implements ServiceListener {
          /*
           * default implementation
           */
      }
    }

I registers service listener and a service. I've got also a different '_http._tcp.' mDNS service running on an ESP wi-fi module. This program detects the service started by itself, but does not detect any other service in the same local network. I replaced `InetAddress.getLocalHost()` with `getIpAddress()`, and it used to work, but now it doesn't.

I even installed [this app](https://play.google.com/store/apps/details?id=com.druk.servicebrowser) on my mobile, and it detects all the services (from laptop and from ESP). I tried to add a service in this app, and JmDNS still doesn't detect it. What am I doing wrong?

**Comment by leif81** [comment]
I see two possible problems with your code there. 

First, your `SampleListener` doesn't do anything. Try overriding the methjods like in the project `README` so that it prints a message.

Second, your main method is going to exit immediately after you call `registerService`. Try adding another sleep after that.

**Comment by aleksandra-majchrzak** [comment]
My `SampleListener` does a few things, but I commented them as, they're not essential for this question. I have solved my problem by... turning my computer off and on again... It turned out that JmDNS stops finding devices after you change WiFi networks - when I connected to different network, and got back to the old one, nothing was found. When I turned my laptop off and on again, everything started working again. This behaviour repeats, so I guess this is how it works. However, it would save me a whole day of struggling, if this information could be found anywhere.
In the end, the problem is solved, so I'm closing the thread.
